source/common/http/conn_manager_impl.cc

```diff
@@ -37,6 +37,31 @@
 namespace Envoy {
 namespace Http {
 
+namespace {
+
+template <class T> using FilterList = std::list<std::unique_ptr<T>>;
+
+// Shared helper for recording the latest filter used.
+template <class T>
+void recordLatestDataFilter(const typename FilterList<T>::iterator current_filter,
```

Comment:
cc @soya3129 this is similar to what you had been doing in one of your metadata PRs in case you end up needing this again.

Reply:
Very nice! Thanks! Can be very useful when we allow metadata to go through downstream filters only.